runtime/metricsview/executor_rewrite_approx_comparisons.go

```diff
@@ -90,6 +90,19 @@ func (e *Executor) rewriteApproxComparisonNode(a *AST, n *SelectNode) bool {
 	}
 	order := []OrderFieldNode{sortField}
 
+	cteOptimization := true
+	if len(n.FromSelect.DimFields) != len(n.JoinComparisonSelect.DimFields) {
+		// generally this should not happen
+		cteOptimization = false
+	}
+	for i := range n.FromSelect.DimFields {
+		if n.FromSelect.DimFields[i].Name != n.JoinComparisonSelect.DimFields[i].Name {
+			// generally this should not happen
+			cteOptimization = false
+			break
+		}
+	}
```

Comment:
Did you find cases where the `DimFields` are not equal for a comparison join? In general, it should not be possible, and it is a requirement that sub-nodes always have all the dimension fields that a parent node has (it may have more for rollups, but never fewer).

So this might better be replaced with a comment or assertion because it should never happen.

Reply:
no I did not but this is just a defensive check, should I replace it with assertion or just remove these checks ?

Reply:
Let's remove the checks, if the `DimFields` don't match, it would cause issues several other places as well. I'll send a PR describing the guarantees better